modules/core/src/smithy4s/http/RawErrorResponse.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ *  Copyright 2021-2024 Disney Streaming
+ *
+ *  Licensed under the Tomorrow Open Source Technology License, Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     https://disneystreaming.github.io/TOST-1.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package smithy4s.http
+
+case class RawErrorResponse(
+    code: Int,
+    headers: Map[CaseInsensitive, Seq[String]],
+    body: String,
+    private val failedDecodeAttemptField: FailedDecodeAttempt
```

Comment:
**issue:** @GaryAghedo I think the intention was to make the whole case class's inner structure private as the linked page suggests. The constructor/extractor will have to become `private`, we'd have `withXXX` methods (immutable setters) instead of `copy`, and we should also probably compile under `-Xsource:3` even on Scala 2 if we're not doing that yet.

Reply:
> we should also probably compile under -Xsource:3 even on Scala 2 if we're not doing that yet.

So currently we are not doing that. 

There was apparently an attempt to enable it in https://github.com/disneystreaming/smithy4s/pull/555/files#r1006283612 but it is unclear why it was left commented out. 

Do you know what is the reason behind following statement from the bincompat guide:
> in Scala 2, you have to add the compiler option `-Xsource:3`

?